.github/workflows/onRelease.yml

```diff
@@ -15,19 +15,38 @@ jobs:
   getDistTag:
     outputs:
       tag: ${{ steps.distTag.outputs.tag }}
+      branch: ${{ steps.getBranch.outputs.branch }}
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
         with:
           ref: ${{ github.event.release.tag_name || inputs.tag  }}
       - uses: salesforcecli/github-workflows/.github/actions/getPreReleaseTag@main
         id: distTag
+      - name: Determine branch
```

Comment:
I don't think you should be able to publish from anything except a github tag on a release (which could be prerelease).  It would simplify a lot of this branch complexity.

Then the content of that tag helps publish know what to do.

Reply:
probably worth imitating https://github.com/forcedotcom/sfdx-core/actions/runs/16426625939/workflow